plugin/adsk/scripts/AEmayaUsdProxyShapeBaseTemplate.mel

```diff
@@ -56,6 +56,75 @@ global proc AEmayaUsdProxyShapePayloadsReplace(string $loadPayloads)
+    columnLayout -adj true;
+        rowLayout -nc 4 -adj 2;
+            text -label `uiRes("m_scriptEditorPanel.kFile")` -ann `getMayaUsdString("kFileAnn")`;
+            textField mayaUsdProxyShapeFilePath;
+            symbolButton -image "navButtonBrowse.png" mayaUsdProxyShapeFilePathBrowser;
+            symbolButton -image "refresh.png" mayaUsdProxyShapeFilePathRefresh;
+        setParent ..;
+    setUITemplate -ppt;
+
+    AEmayaUsdProxyShapeFilePathReplace($filePathAttr);
+}
+
+global proc AEmayaUsdProxyShapeFilePathReplace(string $filePathAttr)
+{
+    // We don't want to connect FilePath control to the filePath attribute because we
+    // want to perform special validation on input before loading any new file.
+    // So do NOT use "connectControl" command.
+
+    string $currFilePath = `getAttr $filePathAttr`;
+    textField -e -fileName $currFilePath -changeCommand ("AEmayaUsdProxyShapeFilePathField(\"" + $filePathAttr + "\"\, \"" + $currFilePath + "\")") mayaUsdProxyShapeFilePath;
+    symbolButton -e -c ("AEmayaUsdProxyShapeFilePathBrowser(\"" + $filePathAttr + "\")") mayaUsdProxyShapeFilePathBrowser;
+    symbolButton -e -c ("AEmayaUsdProxyShapeFilePathRefresh(\"" + $filePathAttr + "\")") -enable `size($currFilePath)` mayaUsdProxyShapeFilePathRefresh;
+
+    // Add a script job on the filepath attribute so we can update the AE control
+    // if the attribute changes from the outside.
+    global int $mayaUsdProxyShapeFilePathScriptJobId = -1;
+    if (`scriptJob -exists $mayaUsdProxyShapeFilePathScriptJobId`)
+        scriptJob -kill $mayaUsdProxyShapeFilePathScriptJobId;
+    string $cmd = "AEmayaUsdProxyShapeFilePathChanged(\"" + $filePathAttr + "\")";
+    $mayaUsdProxyShapeFilePathScriptJobId = `scriptJob -parent mayaUsdProxyShapeFilePath -attributeChange $filePathAttr $cmd`;
+}
+
+// Called with file path attribute text field changes.
+global proc AEmayaUsdProxyShapeFilePathField(string $filePathAttr, string $currFilePath)
+{
+    string $newFilePath = `textField -q -fileName mayaUsdProxyShapeFilePath`;
+    int $result = python("import AETemplateHelpers; AETemplateHelpers.ProxyShapeFilePathChanged('" + $filePathAttr + "', newFilePath='" + $newFilePath + "')");
+    if ($result == 0) {
+        // The proxy shape file path update was rejected (no setAttr was done).
+        // Reset the field back to its original value.
+        textField -e -fileName $currFilePath mayaUsdProxyShapeFilePath;
+    }
+}
+
+// Called when user clicks the file path browser button.
+global proc AEmayaUsdProxyShapeFilePathBrowser(string $filePathAttr)
+{
+    python("import AETemplateHelpers; AETemplateHelpers.ProxyShapeFilePathChanged('" + $filePathAttr + "')");
+}
+
+// Called when the user clicks the file path refresh button.
+global proc AEmayaUsdProxyShapeFilePathRefresh(string $filePathAttr)
+{
+    python("import AETemplateHelpers; AETemplateHelpers.ProxyShapeFilePathRefresh('" + $filePathAttr + "')");
+}
```

Comment:
Wrapped these in functions (instead of attaching the python code directly to the buttons) so that the user will see a better name, rather than obscure python code.